packages/core/src/Modules/Cashier/Config/on-ramp-providers.js

```diff
@@ -0,0 +1,133 @@
+import { getCurrencyDisplayCode } from '@deriv/shared';
+import { localize } from '@deriv/translations';
+import { WS } from 'Services';
+
+// TODO: Write tests to enforce provider structure.
+
+const createBanxaProvider = () => ({
+    icon: { dark: 'IcCashierBanxaDark', light: 'IcCashierBanxaLight' },
+    name: 'Banxa',
+    getDescription: () =>
+        localize(
+            'A fast and secure fiat-to-crypto payment service. Deposit cryptocurrencies from anywhere in the world using your credit/debit cards and bank transfers.'
+        ),
+    getAllowedResidencies: () => ['*'],
+    getPaymentIcons: () => [
+        { dark: 'IcCashierVisaDark', light: 'IcCashierVisaLight' },
+        { dark: 'IcCashierMastercardDark', light: 'IcCashierMastercardLight' },
+        { dark: 'IcCashierBpayDark', light: 'IcCashierBpayLight' },
+        { dark: 'IcCashierSepaDark', light: 'IcCashierSepaLight' },
+        { dark: 'IcCashierBlueshyftDark', light: 'IcCashierBlueshyftLight' },
+        { dark: 'IcCashierFlexepinDark', light: 'IcCashierFlexepinLight' },
+        { dark: 'IcCashierPayIdDark', light: 'IcCashierPayIdLight' },
+        { dark: 'IcCashierPoliDark', light: 'IcCashierPoliLight' },
+        { dark: 'IcCashierApplePay', light: 'IcCashierApplePay' },
+        { dark: 'IcCashierInteracEtransfer', light: 'IcCashierInteracEtransfer' },
+        { dark: 'IcCashierIdeal', light: 'IcCashierIdeal' },
+        { dark: 'IcCashierPostBillPayDark', light: 'IcCashierPostBillPayLight' },
+        { dark: 'IcCashierSofort', light: 'IcCashierSofort' },
+    ],
+    getScriptDependencies: () => [], // None.
+    getDefaultFromCurrency: () => '',
+    getFromCurrencies: () => ['*'],
+    getToCurrencies: () => ['*'],
+    getWidgetHtml: () => {
+        return new Promise((resolve, reject) => {
+            WS.serviceToken({
+                service_token: 1,
+                service: 'banxa',
+                referrer: window.location.href,
+            }).then(response => {
+                if (response.error) {
+                    reject(response.error.message);
+                } else {
+                    const { url } = response.service_token.banxa;
+
+                    if (url) {
+                        window.open(url);
```

Comment:
How come this is needed?

Reply:
We're opening a new tab/window for the user as the process takes place off-site. Is there another way to approach this?

Reply:
No i'm just curious as to how it works 🙂